### DISCUSSION TITLE
Publii na Internet Computer blockchain

### DISCUSSION THREAD

**Discussion by contano** [GetPublii/Publii#1350] [Ideas]
ENG
Publii on Internet Computer blockchain

I know that maybe this is too crazy to take up this topic, but I think that if there was a possibility of easy, direct deployment of Publii on ICP, it would be a great innovation for such a good CMS as Publii. The possibility of such deployment exists at the moment through the Fleek service. The process looks like this: you have to upload Publii to GitHub and then choose Fleek deployment via Internet Computer. In this whole process there are two intermediate steps, namely GitHub and Fleek. My idea is that Publii could be deployed directly to ICP, bypassing the intermediate steps, without using GitHub, straight from the Publii interface. Unfortunately, I am not a programmer. I am a fairly savvy user and I have several of my websites on Publii, creating a photography service. I had everything deployed by Fleek on ICP, but something broke and the changes stopped syncing. They told me they knew about the problems and advised me to switch to IPFS for now and I did. I would love to deploy my sites directly to ICP, if it was as simple as what Publii currently offers with other deployment options through the Publii interface. I am a big fan of ICP technology and I think it is an incredibly innovative and futuristic solution for the Internet in general. I hope I’m not cluttering up the discussion with this idea :) I know you might think it’s an irrational idea :) but maybe it’s worth looking into? I write from the point of view of a fairly advanced user who likes technology. Unfortunately, I don’t have the skills to take up this topic, but you probably do :)

PL
Wiem, że być może jest to zbyt szalone, aby podjąć ten temat, ale uważam, że gdyby istniała możliwość łatwego, bezpośredniego wdrożenia Publii na ICP, byłoby to świetną innowacją dla tak dobrego CMS jakim jest Publii. Możliwość takiego wdrożenia istnieje w chwili obecnej za pośrednictwem usługi Fleek. Proces wygląda w taki sposób, że trzeba wgrać Publii na GitHub a następnie wybrać na Fleek wdrożenie poprzez Internet Computer. W całym tym procesie są dwa kroki pośrednie, czyli GitHub i Fleek. Moja idea polega na tym, żeby Publii mogło być wdrażane od razu na ICP z pominięciem kroków pośrednich, bez konieczności użytkowania GitHub, prosto z interfejsu Publii. Nie jestem niestety programistą. Jestem dość rozgarniętym użytkownikiem i mam na Publii kilka swoich stron internetowych, tworzących serwis o fotografii. Miałem wszystko wdrożone przez Fleek na ICP, ale coś im się zepsuło i zmiany przestały się synchronizować. Powiedzieli mi, że wiedzą o problemach i doradzili, żebym przerzucił się póki co na IPFS i tak zrobiłem. Z wielką ochotą wdrożyłbym moje strony bezpośrednio na ICP, gdyby było to tak proste jak to, co obecnie oferuje Publii z innymi opcjami wdrożenia poprzez interfejs Publii. Jestem wielkim fanem technologii ICP i uważam, że jest to niesamowicie innowacyjne i przyszłościowe rozwiązanie dla Internetu w ogóle. Mam nadzieję, że tą ideą nie zaśmiecam dyskusji :) Wiem, że możecie uznać to za irracjonalny pomysł :) ale może jednak warto się nad nim pochylić? Pisze z punktu wodzenia dość zaawansowanego użytkownika, lubiącego technologię. Nie mam niestety umiejętności, aby podjąć taki temat, ale wy zapewne je macie :)

**Comment by dziudek** [comment]
🇬🇧 Currently, if something can be done - even indirectly as in this case - that is good enough for us :) We have such a long list of important things to implement that niche technologies like IPFS, IPC definitely have to wait. And here de facto there is no problem of use - it's just a question of simplifying it. So it is not a critical problem.

Of course, if it turns out that there is a high demand for this type of functionality, they will get a higher priority in our queue, but to be honest, as much as I have looked at IPFS, I have heard about ICP for the first time - it is rather a solution known to blockchain technology enthusiasts at the moment. And there is a problem with this technology, too, as there are increasing claims that it is highly inefficient. But that's a topic for a completely different discussion.

Certainly, if more Publia users show interest in around-blockchain solutions, we will consider addressing this topic. For the moment, we have many more priority tasks strongly anticipated by the Publia community

P.S For wider availability of posts, I suggest always adding an English translation - e.g. via DeepL :)

------------------------------

🇵🇱 Aktualnie, jeśli da się coś zrobić - nawet pośrednio tak jak w tym wypadku - to jest to dla nas wystarczające rozwiązanie :) Mamy tak długa listę ważnych rzeczy do wdrożenia, że niszowe technologie typu IPFS, IPC zdecydowanie muszą poczekać. A tu de facto nie ma problemu z użyciem - jest kwestia tylko uproszczenia tego. Zatem nie jest to krytyczny problem.

Oczywiście jeśli się okaże, że istnieje duże zapotrzebowanie na tego typu funkcje, to otrzymają wyższy priorytet w naszej kolejce, ale prawdę mówiąc tak jak IPFS się przyglądałem tak o ICP pierwsze słyszę - raczej jest to na ten moment rozwiązanie znane pasjonatom technologii blockchain. A z tą technologią też jest problem, bo coraz częściej spotyka się głosy, że jest mocno nieefektywna. Ale to temat na zupełnie inną dyskusję.

Z pewnością, jeśli więcej userów Publii będzie wykazywać zainteresowanie około-blockchainowymi rozwiązaniami, to rozważymy zaadresowanie tego tematu. Na ten moment mamy dużo bardziej priorytetowych zadań mocno oczekiwanych przez społeczność Publii

P.S Dla szerszej dostępności wpisów, sugeruję zawsze dodawać tłumaczenie na angielski na początku - np. poprzez DeepL :)

**Reply by contano** [reply]
ENG
Thank you for the information about English. I translated the post. I understand the higher priorities. However, I think that in your free time it is worth looking at the ICP technology, because it is something completely different from the traditional understanding of blockchain. Using the vast majority of known blockchain technologies for such a purpose as deploying CMS would be completely pointless in my opinion. ICP opens up completely new possibilities, because it is a different approach. For now, my Publii pages work very well after deployment on IPFS and maybe someday in the future there will be possibilities to deploy directly on ICP :)

PL
Dziękuję za informację o angielskim. Przetłumaczyłem wpis. Rozumiem ważniejsze priorytety. Myślę jednak, że w wolnym czasie warto spojrzeć na technologię ICP, gdyż ona jest czymś zupełnie innym niż tradycyjne rozumienie blockchain. Użycie zdecydowanej większości znanych technologii blockchain do takiego celu jak wdrażanie CMS byłoby moim zdaniem kompletnie bez sensu. ICP otwiera zupełnie nowe możliwości, bo to jest inne podejście. Póki co moje strony Publii bardzo dobrze działają po wdrożeniu na IPFS i może kiedyś w przyszłości pojawią się możliwości wdrożenia bezpośrednio na ICP :)
